src/MobileRedirection.php

```diff
@@ -481,10 +526,12 @@ private function sanitize_script_attributes( $attributes ) {
 	 * @link https://developers.google.com/search/mobile-sites/mobile-seo/separate-urls#annotation-in-the-html
 	 */
 	public function add_mobile_alternative_link() {
-		printf(
-			'<link rel="alternate" type="text/html" media="only screen and (max-width: 640px)" href="%s">',
-			esc_url( $this->get_current_amp_url() )
-		);
+		if ( ! amp_is_request() ) {
```

Comment:
I just found a hole. If someone disables AMP for a given post, at the moment the links are still added when they shouldn't be.

```suggestion
		if ( amp_is_available() && ! amp_is_request() ) {
```